balena-docs/secure-boot.md

```diff
@@ -0,0 +1,108 @@
+
+Balena suggests starting your exploration with Intel NUC devices or ASUS NUC devices. These devices work easily with secure boot and full disk encryption. More specifically we recommend the Intel NUC 11 Essentials models and Intel NUC 13 Pro models. As noted above, please use the `Generic x86_64 (GPT)` device type with your hardware.
+
+## Using secure boot and full disk encryption
+There are two steps required to install a secure boot enabled and disk encrypted system:
+
+* Opt-in secure boot mode in the balenaOS installer
+* Configure the device’s Unified Extensible Firmware Interface (UEFI) firmware for secure boot and setup mode
+
+Note that balenaOS currently does not support updating from a non-secure boot enabled system into a secure boot enabled one. The only way to install a secure boot and disk encrypted system at this moment is by using a balenaOS installer/flasher image.
+
+### Configuring balenaOS installer to opt-in secure boot mode
+Use the [configure](https://docs.balena.io/reference/balena-cli/#os-configure-image) command in balenaCLI (versions >= 16.2.0) to opt-in to secure boot:
+
+```
+balena os configure <image> --secureBoot --version <version> --fleet <fleetName> --device-type <deviceType>
+```
+
+This results in a change to config.json:
+
+```
+"installer": {
+  "secureboot": true
+}
+```
+
+### Configuring the device’s UEFI firmware
+Once the balenaOS image is ready, the device needs to be configured in secure boot setup mode. This depends on the UEFI implementation, but in general there are the following steps to consider:
+
+* Reset to the default UEFI configuration.
+* Make sure the device is configured to boot in UEFI mode, for example by checking the Compatibility Support Module (CSM) used for Master Boot Record (MBR) booting is disabled.
+* Change the boot options to allow booting only from the USB installer/flasher device and the main storage, and choose the USB as first boot option. The flasher will then set the UEFI configuration to boot from the main storage before rebooting.
+* Disable restoring factory keys. Some systems default to restoring factory (Microsoft) keys rather than using the balena keys that will be installed during setup.
+* Enable secure boot.
+* Reset secure boot to setup mode
+
+On booting in setup mode, the installer will enroll the keys into UEFI variables and encrypt the disks using the TPM device. Note that enrolling the keys manually via the UEFI setup application, while possible on some systems, is not currently supported by balenaOS as the installer’s bootloader is not signed.
+
+If you are using balenaOS earlier than 5.3.15, an important note is that the first boot after programming is not fully secured yet as the UEFI settings (and the boot order in particular) will only be locked down during the first boot after installation. Make sure to also perform the initial boot after programming with the device in a secure controlled environment.
```

Comment:
That might be better formatted as a callout if that's supported.

Reply:
@alexgg Ross had the same doubt and I also recommended we do a callout for it. Is it supported?